src/test/java/com/solvd/AbstractTest.java

```diff
@@ -27,20 +28,24 @@ protected WebDriver getDriver() {
         return driver;
     }
 
+    @Parameters("browser")
     @BeforeMethod
-    public void setUp() {
-        ChromeOptions options = new ChromeOptions();
-        options.addArguments("start-maximized");
-        options.addArguments("--auto-open-devtools-for-tabs");
+    public void setUp(String browser) {
         WebDriver driver = null;
         try {
-            Properties properties = new Properties();
-            try (InputStream inStream = Main.class.getResourceAsStream("/application.properties")) {
-                properties.load(inStream);
-            } catch (IOException e) {
-                throw new RuntimeException("Failed to load application properties");
+            Properties properties = PropertiesLoader.getProperties();
+            if (browser.equalsIgnoreCase("chrome")) {
+                ChromeOptions chromeOptions = new ChromeOptions();
+                chromeOptions.addArguments("start-maximized");
+                chromeOptions.addArguments("--auto-open-devtools-for-tabs");
+                driver = new RemoteWebDriver(new URL(properties.getProperty("driverUrl")), chromeOptions);
+
```

Comment:
add variable and reuse it in both side for chrome and for firefox    URL remoteUrl = new URL(properties.getProperty("driverUrl"));